src/main/java/byu/Ui.java

```diff
@@ -0,0 +1,55 @@
+package byu;
+
+import java.util.Scanner;
+
+import exceptions.DukeException;
+
+/**
+ * Represents a user interface that deals with the interaction with users.
+ */
+public class Ui {
+
+    private final Scanner sc = new Scanner(System.in);
+    private final String logo = "*\\(^o^)/*";
+    private String output;
+
+    /**
+     * Prints the welcome message.
+     */
+    public void showWelcome() {
```

Comment:
I like how the function are named accurately and at a sufficient level of detail.